### ISSUE TITLE
XLSR53-large feature extraction

### ISSUE THREAD

**Issue by pyyush** [pytorch/fairseq#3611]
#### What is your question?
Is the below code apt for extracting features from XLSR? Also, do you recommend normalizing the waveform if features will be used for downstream speech tasks i.e. Speech Recognition, Speech Translation?

#### Code

```
model, cfg, task = fairseq.checkpoint_utils.load_model_ensemble_and_task(['xlsr_53_56k.pt'])
model = model[0]
model.eval()
feat = model(waveform, mask=False, features_only=True, layer=14)["x"]
```

**Comment by alexeib** [comment]
you should normalize the waveform if you are using a model trained on normalized input (i.e. cfg.task.normalize == true). for the asr finetuning we check this explicitly and raise an error if it doesnt match

otherwise the code seems like it should work :) theres an extract features script in wav2vec-u (unsupervised) that extracts features from any w2v2 model: https://github.com/pytorch/fairseq/blob/master/examples/wav2vec/unsupervised/scripts/wav2vec_extract_features.py

**Comment by pyyush** [comment]
Which one would you recommend - finetune the entire model on downstream tasks or use frozen/extracted features (ie. features from layer 15 as in wav2vec-u) as input to a small model for downstream tasks?
